src/core/load-fragment.c

```diff
@@ -1591,21 +1591,30 @@ int config_parse_service_sockets(
         assert(rvalue);
         assert(data);
 
-        FOREACH_WORD_QUOTED(word, l, rvalue, state) {
-                _cleanup_free_ char *t = NULL, *k = NULL;
+        prev = cur = rvalue;
+        for(;;) {
+                _cleanup_free_ char *word = NULL, *t = NULL, *k = NULL;
 
-                t = strndup(word, l);
-                if (!t)
+                r = extract_first_word(&cur, &word, NULL, 0);
+                if (r == 0)
+                        break;
+                if (r == -ENOMEM)
                         return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Trailing garbage in sockets, ignoring: %s", prev);
```

Comment:
> Well, no, cur is the current pointer, and extracting a word from that failed if you get here. Hence, logging cur is sufficient. No need to preserve the former pointer.

`extract_first_word` [changes](https://github.com/systemd/systemd/blob/b11d6a7bed4d867fb9f6ff4e7eb4ab20fcdc9301/src/basic/extract-word.c#L170) `cur`. so, I can have `-EINVAL` and shifted `cur`.